netbout-web/src/main/java/com/netbout/rest/MarkdownTxtmark.java

```diff
@@ -47,10 +47,18 @@
         // @checkstyle LineLengthCheck (1 line)
         "(?<!\\]\\()(?<!=\")(https?:\\/\\/[a-zA-Z0-9-._~:\\?#@!$&'*+,;=%\\/]+[a-zA-Z0-9-_~#@$&'*+=%\\/])(?![\\w.]*\\]\\()"
     );
+    /**
+     * Pattern to detect lines which should have a line break on the end.
+     */
+    private static final Pattern NEW_LINE = Pattern.compile(
+        "^ {0,3}\\S* ?$", Pattern.MULTILINE
```

Comment:
@dmzaytsev it is still not clear for me what are expectations of this pattern. It matches all of those:
* empty string
* one to four spaces
* any non-whitespace characters, that are surrounded by at most three spaces from left and at most one space on right, so
 * `   a ` is matched
 * `   a a ` is not matched - why?